src/main/java/org/embulk/input/marketo/delegate/CustomObjectInputPlugin.java

```diff
@@ -0,0 +1,75 @@
+package org.embulk.input.marketo.delegate;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.FluentIterable;
+import org.apache.commons.lang3.StringUtils;
+import org.embulk.base.restclient.ServiceResponseMapper;
+import org.embulk.base.restclient.record.ServiceRecord;
+import org.embulk.base.restclient.record.ValueLocator;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.config.ConfigException;
+import org.embulk.input.marketo.MarketoService;
+import org.embulk.input.marketo.MarketoServiceImpl;
+import org.embulk.input.marketo.MarketoUtils;
+import org.embulk.input.marketo.rest.MarketoRestClient;
+import org.embulk.spi.Exec;
+import org.slf4j.Logger;
+import java.util.Iterator;
+
+public class CustomObjectInputPlugin extends MarketoBaseInputPluginDelegate<CustomObjectInputPlugin.PluginTask>
+{
+    private final Logger logger = Exec.getLogger(getClass());
+
+    public interface PluginTask extends MarketoBaseInputPluginDelegate.PluginTask, CustomObjectResponseMapperBuilder.PluginTask
+    {
+        @Config("custom_object_filter_type")
+        @ConfigDefault("\"\"")
+        String getCustomObjectFilterType();
+
+        @Config("custom_object_filter_from_value")
+        @ConfigDefault("1")
+        Integer getFromValue();
+
+        @Config("custom_object_filter_to_value")
+        @ConfigDefault("null")
+        Optional<Integer> getToValue();
+    }
+
+    public CustomObjectInputPlugin()
+    {
+    }
+
+    @Override
+    public void validateInputTask(PluginTask task)
+    {
+        super.validateInputTask(task);
+        if (StringUtils.isBlank(task.getCustomObjectFilterType())) {
+            throw new ConfigException("`custom_object_filter_type` cannot empty");
+        }
+        if (StringUtils.isBlank(task.getCustomObjectAPIName())) {
+            throw new ConfigException("`custom_object_api_name` cannot empty");
```

Comment:
grammatically like above

Reply:
we cannot sure that user will not put empty string on console GUI. I checked and see that if user put empty string in required field , it'll valid. So i check to verify for this case.

Reply:
grammatically like above